PowerShellTools/DebugEngine/ScriptDebugger.cs

```diff
@@ -262,19 +284,27 @@ private void RefreshCallStack()
             IEnumerable<CallStack> result = null;
             try
             {
-                result = DebuggingService.GetCallStack();
+                if (IsDebugging)
+                {
+                    result = DebuggingService.GetCallStack();
+                }
+                else
+                {
+                    throw new Exception("Not in debugging mode any more!");

```

Comment:
I think I was just a bit lazy that trying to use this exception as a fast-fail for this method, and meanwhile leveraging it as a return value to its consumer, so this ex is completely internal and string won't be user-visible. But I agree with you that needs to be more specific exception.